assets/src/edit-story/components/library/panes/media/media3p/media3pPane.js

```diff
@@ -35,19 +33,88 @@ const StyledPane = styled(Pane)`
   overflow: hidden;
 `;
 
-const Inner = styled.div`
+const Container = styled.div`
+  overflow: scroll;
   height: 100%;
-  display: grid;
-  grid:
-    'header   ' auto
-    'infinitescroll' 1fr
-    / 1fr;
+  padding: 0 1.5em 0 1.5em;
+  margin-top: 1em;
 `;
 
+/**
+ * Pane that contains the media 3P integrations.
+ *
+ * @param {Object} props Component props
+ * @return {*} The media pane element for 3P integrations.
+ */
 function Media3pPane(props) {
+  // TODO(#1698): Ensure scrollbars auto-disappear in MacOS.
+  // State and callback ref necessary to recalculate the padding of the list
+  //  given the scrollbar width.
+  const [scrollbarWidth, setScrollbarWidth] = useState(0);
+  const refContainer = useRef();
+  const refCallbackContainer = (element) => {
+    refContainer.current = element;
+    if (!element) {
+      return;
+    }
+    setScrollbarWidth(element.offsetWidth - element.clientWidth);
+  };
+
+  // TODO(#2368): get resources from useMedia
+  // TODO(#2368): handle pagination / infinite scrolling
+  const resources = [
+    {
+      id: 1,
+      type: 'image',
+      local: false,
+      alt: 'image alt',
+      mimeType: 'image/jpeg',
+      width: 18,
+      height: 12,
+      src:
+        'https://img.webmd.com/dtmcms/live/webmd/consumer_assets/site_images/article_thumbnails/slideshows/how_to_brush_dogs_teeth_slideshow/1800x1200_how_to_brush_dogs_teeth_slideshow.jpg',
+    },
+    {
+      id: 1,
+      type: 'image',
+      local: false,
+      alt: 'image alt',
+      mimeType: 'image/jpeg',
+      width: 128,
+      height: 72,
+      src:
+        'https://www.sciencemag.org/sites/default/files/styles/article_main_large/public/dogs_1280p_0.jpg?itok=cnRk0HYq',
+    },
+  ];
+
+  // Recalculates padding of Media Pane so it stays centered.
+  // As of May 2020 this cannot be achieved without js (as the scrollbar-gutter
+  // prop is not yet ready).
+  useLayoutEffect(() => {
+    if (!scrollbarWidth) {
+      return;
+    }
+    const currentPaddingLeft = parseFloat(
+      window
+        .getComputedStyle(refContainer.current, null)
+        .getPropertyValue('padding-left')
+    );
+    refContainer.current.style['padding-right'] =
+      currentPaddingLeft - scrollbarWidth + 'px';
+  }, [scrollbarWidth, refContainer]);
+
+  // Callback for when a media element is selected.
+  const onInsert = () => {};
```

Comment:
Use useMemo, to avoid re-rendering all MediaElements of MediaGallery on each render of Media3pPane:
```
const onInsert = useMemo(() => {}, []);
```

Reply:
Sorry, I meant useCallback, not useMemo :)